test/functional/editions_controller_test.rb

```diff
@@ -520,84 +520,167 @@ class EditionsControllerTest < ActionController::TestCase
   end
 
   context "#send_to_publish" do
-    setup do
-      @requester = FactoryBot.create(:user, :govuk_editor, name: "Stub Requester")
-      @edition = FactoryBot.create(:edition, state: "scheduled_for_publishing", publish_at: Time.zone.now + 1.hour)
-      login_as(@requester)
-    end
-
-    context "user has govuk_editor permission" do
+    context "edition is in 'scheduled_for_publishing' state" do
       setup do
-        ScheduledPublisher.stubs(:cancel_scheduled_publishing)
-        PublishWorker.stubs(:perform_async)
+        @requester = FactoryBot.create(:user, :govuk_editor, name: "Stub Requester")
+        @edition = FactoryBot.create(:edition, state: "scheduled_for_publishing", publish_at: Time.zone.now + 1.hour)
+        login_as(@requester)
       end
 
-      should "update the edition status to 'published' and save the comment" do
-        post :send_to_publish, params: {
-          id: @edition.id,
-          comment: "Publishing with immediate effect",
-        }
+      context "user has govuk_editor permission" do
+        setup do
+          ScheduledPublisher.stubs(:cancel_scheduled_publishing)
+          PublishWorker.stubs(:perform_async)
+        end
 
-        assert_equal "Published", flash[:success]
-        @edition.reload
-        assert_equal "published", @edition.state
-        assert_equal "Publishing with immediate effect", @edition.latest_status_action.comment
-        assert_equal @requester.id, @edition.latest_status_action.requester_id
+        should "update the edition status to 'published' and save the comment" do
+          post :send_to_publish, params: {
+            id: @edition.id,
+            comment: "Publishing with immediate effect",
+          }
+
+          assert_equal "Published", flash[:success]
+          @edition.reload
+          assert_equal "published", @edition.state
+          assert_equal "Publishing with immediate effect", @edition.latest_status_action.comment
+          assert_equal @requester.id, @edition.latest_status_action.requester_id
+        end
+
+        should "not update the edition state and render 'send_to_publish' template with an error when an error occurs" do
+          EditionProgressor.any_instance.expects(:progress).returns(false)
+
+          post :send_to_publish, params: {
+            id: @edition.id,
+          }
+
+          assert_template "secondary_nav_tabs/send_to_publish_page"
+          assert_equal "Due to a service problem, the request could not be made", flash[:danger]
+          @edition.reload
+          assert_equal "scheduled_for_publishing", @edition.state
+        end
+
+        should "notify the publishing API to publish the edition" do
```

Comment:
This test has no assertions. Is that right?

Reply:
Yes, that is correct.
**PublishWorker** is a stub here, and it is expecting that the code in the test will call the PublishWorker.perform_asynch method with the edition id as a string. With any other value it would fail. So it works as intended.